src/utils/cache.ts

```diff
@@ -0,0 +1,44 @@
+import LruCache from 'lru-cache';
+
+/**
+ * A generalized cache interface.
+ */
+export interface Cache {
+  /**
+   * Sets a key-value pair.
+   * @returns `true` if key-value pair is stored successfully; `false` otherwise
+   */
+  set(key: string, value: any): Promise<boolean>;
+
+  /**
+   * Gets the value corresponding to the given key.
+   * @returns value stored corresponding to the given key; `undefined` if key is not found or expired
+   */
+  get(key: string): Promise<any | undefined>;
+}
+
```

Comment:
2 Questions:
1. What's the primary motivation behind a generalized `Cache` interface? is it so different caches can be implemented with different underlying stores (e.g. redis) while maintaining the same interface?
2. to remain consistent with where exported types are put in the rest of the project, should we put the `Cache` interface into `utils/types.ts`?

Reply:
1. Yes, this already happened in this PR, and preps for redis.
1. Make sense: 82fcfda